### ISSUE TITLE
Feature Request: Configurable Mouse Cursor Hiding

### ISSUE THREAD

**Issue by davxy** [pop-os/cosmic-settings#603]
It would be great to have an option in the settings to hide the mouse cursor after a configurable time period of inactivity. Additionally, there could be an option to immediately hide the cursor when the user starts typing. This feature is quite popular among users of tiling window managers (such as Sway and Hyprland), where the mouse is often secondary to keyboard navigation.

Use Case:

For users who rely heavily on keyboard input and use tiling, the mouse cursor can often become a distraction when it stays visible on the screen after periods of inactivity. By providing a setting to automatically hide the cursor after a configurable timeout, or immediately when typing begins, the user experience can be significantly improved, especially for those using keyboard-driven workflows.

Proposed Settings:

1. Hide cursor after inactivity:

Type: Boolean (Enable/Disable)
Timeout: Configurable duration (e.g., 5 seconds, 10 seconds, etc.)
Default: Disabled

2. Hide cursor when typing:

Type: Boolean (Enable/Disable)
Default: Disabled

Thank you for considering this request!

**Comment by awilkins** [comment]
+1 for this : 

Especially considering that moving focus with the keys also brings the mouse pointer along with it and parks it centre screen, right in the middle of e.g. your multi-pane `vim` window.

**Comment by el4ra** [comment]
+1
Heavy user of tiling with COSMIC on NixOS. Would love the ability to hide the cursor (or at least keep it in place) especially when jumping from screen to screen using Super+Arrows. Right now the cursor will go to the center of the screen and stay there.

**Comment by virackson** [comment]
Is there any possibility of this feature being implemented? The inactive mouse cursor tends to be a bit distracting, and I often find myself needing to move it manually to some corner. Thank you so much for considering!

**Comment by cureforboredom** [comment]
+1

I can deal with the jank and bugs, that honestly doesn't annoy me and I am loving cosmic.

I just need my mouse cursor to get out of the way when I'm or otherwise not using it. For whatever reason, VSCode doesn't hide the cursor, which is incredibly frustrating, but on X I could use xbanish. Please consider this feature.

**Comment by mmj2023** [comment]
Yeah, all other wayland compositors that I have tried have the option to auto hide the cursor. Hopefully, this gets added soon. But, is there a manual way like changing the config or running a command to set the option?

**Comment by wezm** [comment]
It looks like this was implemented in https://github.com/pop-os/cosmic-comp/pull/2393 but there is no UI to control it yet. You can manually set the config value though:

```
echo 'Some(3)' > ~/.config/cosmic/com.system76.CosmicComp/v1/cursor_hide_timeout
```

This makes it hide after 3 seconds.
